horovod/common/ops/nccl_operations.h

```diff
@@ -38,15 +38,12 @@ struct NCCLContext {
   void ShutDown();
 };
 
-class NCCLAllreduce : public CUDAAllreduce {
+class NCCLOp {
 public:
-  NCCLAllreduce(NCCLContext* nccl_context, CUDAContext* cuda_context,
-                HorovodGlobalState* global_state)
-      : CUDAAllreduce(cuda_context, global_state),
-        nccl_context_(nccl_context){};
-
-  Status Execute(std::vector<TensorTableEntry>& entries,
-                 const Response& response) override;
+  NCCLOp(NCCLContext* nccl_context, HorovodGlobalState* global_state)
```

Comment:
In a previous PR (https://github.com/horovod/horovod/pull/1480/files#diff-5798de9e6fdc83e7f7bf0fa8344a56c0R65) we opted to use delegation to solve this problem by introducing `CUDAOpContext`.  It looks like we could do something similar here, which would make things in our codebase more consistent.  What do you think?

Reply:
I agree to try to keep things consistent through the code base. Let me give it a try.